.gitignore

```diff
@@ -31,6 +32,16 @@
 /redex.tar.gz
 /redexdump
 /stamp-h1
+/test/.libs/
+/test/googletest-release-*/
+/test/gtest-*.zip
+/test/setup_gtest
+/test/**/*.log
+/test/**/*.trs
+/test/**/*_test
+/test/**/*.jar
+/test/**/*.dex
```

Comment:
We should probably be careful about this. If we add a test case with checked in jars or dexes, this gitignore will cause them to be omitted from the repo. Which test case did you find these files?

Reply:
I did a grep on a clean master `check` run.
test/integ/ab_experiment_context_test-class.dex
test/integ/call_graph_test-class.dex
test/integ/clinit_side_effect_test-class.dex
test/integ/constant_propagation_test-class.dex
test/integ/constructor_dedup_test_test-class.dex
test/integ/copy_prop_test-class.dex
test/integ/dedup_blocks_test-class.dex
test/integ/dedup_vmethods_test-class.dex
test/integ/default_annotation_test-class.dex
test/integ/dex_output_test-class.dex
test/integ/final_inline_analysis_test-class.dex
test/integ/global_type_analysis_test-class.dex
test/integ/instruction_sequence_outliner_test-class.dex
test/integ/iodi_test-class.dex
test/integ/ip_reflection_analysis_test-class.dex
test/integ/max_depth_test-class.dex
test/integ/method_override_graph_test-class.dex
test/integ/monotonic_fixpoint_test-class.dex
test/integ/peephole_test-class.dex
test/integ/points_to_semantics_test-class.dex
test/integ/propagation_test-class.dex
test/integ/pure_method_test-class.dex
test/integ/reachability_test-class.dex
test/integ/reflection_analysis_test-class.dex
test/integ/remove_unreachable_test-class.dex
test/integ/result_propagation_test-class.dex
test/integ/strip_debug_info_test-class.dex
test/integ/synth_test-class.dex
test/integ/type_analysis_transform_test-class.dex
test/integ/type_inference_test-class.dex

test/integ/ab_experiment_context_test-class.jar
test/integ/ab_experiment_context_test-class.jar.tmp/
test/integ/call_graph_test-class.jar
test/integ/call_graph_test-class.jar.tmp/
test/integ/clinit_side_effect_test-class.jar
test/integ/clinit_side_effect_test-class.jar.tmp/
test/integ/constant_propagation_test-class.jar
test/integ/constant_propagation_test-class.jar.tmp/
test/integ/constructor_dedup_test_test-class.jar
test/integ/constructor_dedup_test_test-class.jar.tmp/
test/integ/copy_prop_test-class.jar
test/integ/copy_prop_test-class.jar.tmp/
test/integ/dedup_blocks_test-class.jar
test/integ/dedup_blocks_test-class.jar.tmp/
test/integ/dedup_vmethods_test-class.jar
test/integ/dedup_vmethods_test-class.jar.tmp/
test/integ/default_annotation_test-class.jar
test/integ/default_annotation_test-class.jar.tmp/
test/integ/dex_output_test-class.jar
test/integ/dex_output_test-class.jar.tmp/
test/integ/final_inline_analysis_test-class.jar
test/integ/final_inline_analysis_test-class.jar.tmp/
test/integ/global_type_analysis_test-class.jar
test/integ/global_type_analysis_test-class.jar.tmp/
test/integ/instruction_sequence_outliner_test-class.jar
test/integ/instruction_sequence_outliner_test-class.jar.tmp/
test/integ/iodi_test-class.jar
test/integ/iodi_test-class.jar.tmp/
test/integ/ip_reflection_analysis_test-class.jar
test/integ/ip_reflection_analysis_test-class.jar.tmp/
test/integ/max_depth_test-class.jar
test/integ/max_depth_test-class.jar.tmp/
test/integ/method_override_graph_test-class.jar
test/integ/method_override_graph_test-class.jar.tmp/
test/integ/monotonic_fixpoint_test-class.jar
test/integ/monotonic_fixpoint_test-class.jar.tmp/
test/integ/peephole_test-class.jar
test/integ/peephole_test-class.jar.tmp/
test/integ/points_to_semantics_test-class.jar
test/integ/points_to_semantics_test-class.jar.tmp/
test/integ/propagation_test-class.jar
test/integ/propagation_test-class.jar.tmp/
test/integ/pure_method_test-class.jar
test/integ/pure_method_test-class.jar.tmp/
test/integ/reachability_test-class.jar
test/integ/reachability_test-class.jar.tmp/
test/integ/reflection_analysis_test-class.jar
test/integ/reflection_analysis_test-class.jar.tmp/
test/integ/remove_unreachable_test-class.jar
test/integ/remove_unreachable_test-class.jar.tmp/
test/integ/result_propagation_test-class.jar
test/integ/result_propagation_test-class.jar.tmp/
test/integ/strip_debug_info_test-class.jar
test/integ/strip_debug_info_test-class.jar.tmp/
test/integ/synth_test-class.jar
test/integ/synth_test-class.jar.tmp/
test/integ/type_analysis_transform_test-class.jar
test/integ/type_analysis_transform_test-class.jar.tmp/
test/integ/type_inference_test-class.jar
test/integ/type_inference_test-class.jar.tmp/